src/net.py

```diff
@@ -622,11 +646,18 @@ def train(self, epochs, train_x, train_y, val_x, val_y,
                     plt.figure(fig_number)
                     display_record(record=self.record)
 
+
+
         except KeyboardInterrupt:
             print ("\n\n**********Training stopped prematurely.**********\n\n")
         finally:
             self.timestamp = get_timestamp()
 
+            if self.stopping_rule == 'best_validation':
+                print("STOPPING RULE: Rewinding to epoch {} which had the lowest validation error: {}".format(best_epoch, best_error))
+                self.__setstate__(best_state)
```

Comment:
If we were to just save the weights of the network (using my previous recommendation) then we can load them back into the network here with a command similar to ```lasagne.layers.set_all_param_values(self.layers, weights, **{self.name: True})```. I think this would run faster. thoughts?